### PR TITLE
Making testlib shellcheck compliant

### LINKED ISSUE #662: Fix reported warnings from shellcheck in testlib
Shellcheck is reporting some warnings on testlib and we should fix the ones that are relevant. If there are checks that doesn't apply, comment here.

### PR DESCRIPTION
**Pull request by castulo**
Shellcheck provides many useful recommendations for writing shell code.
This commit makes testlib.bash shellcheck compliant.

Signed-off-by: Castulo J. Martinez <castulo.martinez@intel.com>

Closes #662

### CODE REVIEW

File: test/functional/testlib.bash [hunk 1]
```diff
@@ -2928,6 +2924,7 @@ print_assert_failure() {
 	echo -e "$message"
 	echo "Command output:"
 	echo "------------------------------------------------------------------"
+	# shellcheck disable=SC2154
```
Comment (otaviobp) [inline test/functional/testlib.bash]: add a comment why you are disabling this checker

Comment (castulo) [reply test/functional/testlib.bash]: done

**Comment by castulo** [comment]
> lgtm. I prefer to merge this even if it's not fixing all the problems. Than you can have another PR fixing the remaining problems.

I'm almost done fixing all the issues

**Comment by castulo** [comment]
All issues have been fixed

**Pull request merged**